alibi_detect/od/sklearn/base.py

```diff
@@ -0,0 +1,220 @@
+from __future__ import annotations
+from typing import List, Union, Optional, Dict
+from dataclasses import dataclass, asdict
+from abc import ABC, abstractmethod
+from typing_extensions import Self
+
+import numpy as np
+
+from alibi_detect.exceptions import NotFittedError, ThresholdNotInferredError
+
+
+@dataclass
+class SklearnOutlierDetectorOutput:
+    """Output of the outlier detector."""
+    threshold_inferred: bool
+    instance_score: np.ndarray
+    threshold: Optional[np.ndarray]
```

Comment:
Atm, whereas other detectors indeed output the threshold as a singleton array, the sklearn GMM detector outputs a floa. Perhaps a float is better though?

Reply:
Yeah, I agree that the threshold should be kept as just a float. I'm going to defer to a later PR though as it'll make changes to the other detectors.